File: trunk/theLink/example/go/Filter5.go
/**
 *  \file       theLink/example/go/Filter5.go
 *  \brief      \$Id: Filter5.go 357 2010-12-11 10:36:08Z aotto1968 $
 *  
 *  (C) 2010 - NHI - #1 - Project - Group
 *  
 *  \version    \$Rev: 357 $
 *  \author     EMail: aotto1968 at users.berlios.de
 *  \attention  this software has GPL permissions to copy
 *              please contact AUTHORS for additional information
 */

package main

import (
  . "gomsgque"
    "os"
)

// F1 ************************************************************

type F1 struct { *MqS }

func F1New (tmpl *MqS) *MqS {
  srv := new(F1)
  srv.MqS = NewMqS(tmpl, srv)
  return srv.MqS
}

func (this *F1) ServerSetup() {
  this.ServiceCreate("+FTR", this)
  this.ServiceProxy("+EOF")
}

func (this *F1) Call() {
  ftr := this.ServiceGetFilter()
  ftr.SendSTART()
  ftr.SendC("F1")
  ftr.SendC(ftr.ConfigGetName())
  ftr.SendI(int32(ftr.ConfigGetStartAs()))
  for this.ReadItemExists() {
    ftr.SendC(this.ReadC())
  }
  ftr.SendEND_AND_WAIT2("+FTR")
  this.SendRETURN()
}

// F2 ************************************************************

type F2 struct { *MqS }

func F2New (tmpl *MqS) *MqS {
  srv := new(F2)
  srv.MqS = NewMqS(tmpl, srv)
  return srv.MqS
}

func (this *F2) ServerSetup() {
  this.ServiceCreate("+FTR", this)
  this.ServiceProxy("+EOF")
}

func (this *F2) Call() {
  ftr := this.ServiceGetFilter()
  ftr.SendSTART()
  ftr.SendC("F2")
  ftr.SendC(ftr.ConfigGetName())
  ftr.SendI(int32(ftr.ConfigGetStartAs()))
  for this.ReadItemExists() {
    ftr.SendC(this.ReadC())
  }
  ftr.SendEND_AND_WAIT2("+FTR")
  this.SendRETURN()
}

// F3 ************************************************************

type F3 struct { *MqS }

func F3New (tmpl *MqS) *MqS {
  srv := new(F3)
  srv.MqS = NewMqS(tmpl, srv)
  return srv.MqS
}

func (this *F3) ServerSetup() {
  this.ServiceCreate("+FTR", this)
  this.ServiceProxy("+EOF")
}

func (this *F3) Call() {
  ftr := this.ServiceGetFilter()
  ftr.SendSTART()
  ftr.SendC("F3")
  ftr.SendC(ftr.ConfigGetName())
  ftr.SendI(int32(ftr.ConfigGetStartAs()))
  for this.ReadItemExists() {
    ftr.SendC(this.ReadC())
  }
  ftr.SendEND_AND_WAIT2("+FTR")
  this.SendRETURN()
}

// main *********************************************************

func main() {
  FactoryAdd("F1", F1New)
  FactoryAdd("F2", F2New)
  FactoryAdd("F3", F3New)
  srv := FactoryCall(os.Args[1])
  Init(os.Args[0])
  defer func() {
    if x := recover(); x != nil {
      srv.ErrorSet(x)
    }
    srv.Exit()
  }()
  srv.LinkCreate(os.Args[1:]...)
  srv.ProcessEvent(WAIT_FOREVER)
}

File: trunk/theLink/example/vb/Filter5.vb
REM

Imports System
Imports System.IO
Imports csmsgque

Public Module example

  REM F1 ********************************************************
  Private NotInheritable Class F1
    Inherits MqS
    Implements IServerSetup

    ' constructor
    Public Sub New(ByVal tmpl As MqS)
      MyBase.New(tmpl)
    End Sub

    Sub FTR()
      Dim ftr As MqS = ServiceGetFilter()
      ftr.SendSTART()
      ftr.SendC("F1")
      ftr.SendC(ftr.ConfigGetName())
      ftr.SendI(CType(ftr.ConfigGetStartAs(), Integer))
      ftr.SendI(ConfigGetStatusIs())
      While ReadGetNumItems() <> 0
        ftr.SendC(ReadC())
      End While
      ftr.SendEND_AND_WAIT("+FTR")
      SendRETURN()
    End Sub

    Private Sub ServerSetup() Implements IServerSetup.ServerSetup
      ServiceCreate("+FTR", AddressOf FTR)
      ServiceProxy("+EOF")
    End Sub
  End Class

  REM F2 ********************************************************
  Private NotInheritable Class F2
    Inherits MqS
    Implements IServerSetup

    ' constructor
    Public Sub New(ByVal tmpl As MqS)
      MyBase.New(tmpl)
    End Sub

    Private Sub FTR()
      Dim ftr As MqS = ServiceGetFilter()
      ftr.SendSTART()
      ftr.SendC("F2")
      ftr.SendC(ftr.ConfigGetName())
      ftr.SendI(CType(ftr.ConfigGetStartAs(), Integer))
      ftr.SendI(ConfigGetStatusIs())
      While ReadGetNumItems() <> 0
        ftr.SendC(ReadC())
      End While
      ftr.SendEND_AND_WAIT("+FTR")
      SendRETURN()
    End Sub

    Private Sub ServerSetup() Implements IServerSetup.ServerSetup
      ServiceCreate("+FTR", AddressOf FTR)
      ServiceProxy("+EOF")
    End Sub
  End Class

  REM F3 ********************************************************
  Private NotInheritable Class F3
    Inherits MqS
    Implements IServerSetup

    ' constructor
    Public Sub New(ByVal tmpl As MqS)
      MyBase.New(tmpl)
    End Sub

    Private Sub FTR()
      Dim ftr As MqS = ServiceGetFilter()
      ftr.SendSTART()
      ftr.SendC("F3")
      ftr.SendC(ftr.ConfigGetName())
      ftr.SendI(CType(ftr.ConfigGetStartAs(), Integer))
      ftr.SendI(ConfigGetStatusIs())
      While ReadGetNumItems() <> 0
        ftr.SendC(ReadC())
      End While
      ftr.SendEND_AND_WAIT("+FTR")
      SendRETURN()
    End Sub

    Private Sub ServerSetup() Implements IServerSetup.ServerSetup
      ServiceCreate("+FTR", AddressOf FTR)
      ServiceProxy("+EOF")
    End Sub
  End Class

  REM Main ******************************************************
  Sub Main(ByVal args() As String)
    MqFactoryS(Of F1).Add()
    MqFactoryS(Of F2).Add()
    MqFactoryS(Of F3).Add()
    Dim srv As MqS = MqFactoryS(Of MqS).GetCalled(args(0)).[New]()
    Try
      srv.LinkCreate(args)
      srv.ProcessEvent(MqS.WAIT.FOREVER)
    Catch ex As Exception
      srv.ErrorSet(ex)
    End Try
    srv.Exit()
  End Sub
End Module

